app/controllers/api/v1/users_controller.rb

```diff
@@ -3,7 +3,45 @@
 module API
   module V1
     class UsersController < API::V1::APIController
-      before_action :auth_user
+      before_action :auth_user, except: [:impersonate]
+      skip_before_action :authenticate_user!, only: [:impersonate]
+      skip_after_action :verify_authorized, only: :impersonate
+
+      def impersonate
+        # TODO use encryptorService para desencriptar queryParams
+        len = ActiveSupport::MessageEncryptor.key_len
+        salt, encrypted_data = params['queryParams'].split '$$'
+        encrypted_data = Base64.urlsafe_decode64(encrypted_data)
+        key = ActiveSupport::KeyGenerator.new(Rails.application.secrets.secret_key_base).generate_key(salt, len)
+        crypt = ActiveSupport::MessageEncryptor.new(key)
+        decrypted_params = JSON.parse(crypt.decrypt_and_verify(encrypted_data, purpose: 'impersonation'))
+        # ----------------------------------------------- END
+        
+        render json: { error: 'Invalid session' } and return unless warden.authenticated?(:admin_user)
```

Comment:
what is this? I don't think we need to check any authentication at this point, that's what we have the encrypted query param for, right?